curriculum/challenges/english/20-upcoming-python/learn-tree-traversal-by-building-a-binary-search-tree/655a167ea3e96512bf1343ce.md

```diff
@@ -0,0 +1,64 @@
+---
+id: 655a167ea3e96512bf1343ce
+title: Step 20
+challengeType: 20
+dashedName: step-20
+---
+
+# --description--
+
+Inside the `search` method, delete `pass` and call the private helper method `_search` with the following arguments:
+
+- `self.root`: This is the root of the binary search tree. The search starts from the root.
+- `key`: This is the value that the user wants to find in the BST.
+
+Internally, `search` delegates the actual search logic to the private `_search` method that performs the actual recursive search within the binary search tree.
+
+You will define `_search` in the next steps.
```

Comment:
```suggestion
Inside the `search` method, delete `pass` and call the private helper method `_search` with the following arguments. You will define `_search` in the next steps.

- `self.root`: This is the root of the binary search tree. The search starts from the root.
- `key`: This is the value that the user wants to find in the binary search tree.

Internally, `search` delegates the actual search logic to the private `_search` method that performs the actual recursive search within the binary search tree.
```
I would suggest mentioning that the private helper method will be defined in the next steps in the first paragraph.